### DISCUSSION TITLE
Built in graph node

### DISCUSSION THREAD

**Discussion by ghost** [godotengine/godot-proposals#4713] [Polls]
I was trying to work out how to make a graph ui node, so like for instance a node called GraphChart, with different types selectable from a property, the different types could be pie chart, line chart, bar chart, dot chart and so on, but how hard would this be to implement in the engine, I think it would benefit many people and improve the engine overrall, please respond.

**Comment by Calinou** [comment]
This is already available with an add-on: https://github.com/fenix-hub/godot-engine.easy-charts

Charting is rarely needed in games, so I don't think it should be core.

**Reply by ghost** [reply]
Thanks, Cal, but this is more about making Godot Engine as feature packed as possible and to attract as many users as possible and also to actually compete with proprietary competitors. And secondly charting is commonly used in games, some foremost examples including the Farming Simulator series, Age of Empires 3 , 4 and some games are even based around charting like Democracy 4 and Game Dev Tycoon, the examples go on.

**Reply by Calinou** [reply]
Making the engine as full-featured as possible is not a goal if the features don't end up being used often. See [Best practices for engine contributors](https://docs.godotengine.org/en/stable/community/contributing/best_practices_for_engine_contributors.html) for more information.

For features that are not used often, add-ons are often the best way to implement them. This way, not everyone has to pay for the cost of implementing and maintaining the feature.

**Reply by sairam4123** [reply]
The add-on you attached is not being maintained regularly. I've had problems getting it to work, and I still have problems with memory consumption and other problems with it.

A engine supported Graph node would help a ton!

**Reply by Calinou** [reply]
I don't think an add-on not being maintained regularly (or having performance problems that can be ironed out) is a reason in itself to add a feature to core.

**Reply by sairam4123** [reply]
It could be made a 1st party GDExtension instead of a 3rd party addon!

**Reply by Calinou** [reply]
> It could be made a 1st party GDExtension instead of a 3rd party addon!

It can also be a third-party GDExtension. Not all GDExtensions have to be first-party :slightly_smiling_face: 

We should be very careful about what kinds of add-ons we make first-party, as this adds a *lot* of support burden. In the past, we've hosted many repositories under the @godotengine organization that ended up being [poorly](https://github.com/godotengine/collada-exporter) [maintained](https://github.com/godotengine/godot-blender-exporter). There's a certain quality guarantee we make by adding repositories to the @godotengine organization, and we can only spread ourselves so thin.

**Comment by ghost** [comment]
And I think I accidentally voted on my own poll, **OOPS!**

**Comment by azur-wolve** [comment]
Another interesting use I see for that is for testing and analyze, as developer.

**Reply by ghost** [reply]
BAM! Good one, KiwwiPity.

**Comment by ghost** [comment]
This is a close call, this poll!

**Reply by Calinou** [reply]
Please don't bump your own thread without contributing significant new information.

**Reply by ghost** [reply]
Uh, sorry, I am not quite sure how that works, I thought it's kind of like my thread, and does bump mean comment?

**Reply by Calinou** [reply]
Yes. Commenting is "bumping" as it creates notifications for everyone subscribed to the thread (which is, anyone who replied).

**Reply by ghost** [reply]
OK, I understand, thanks for explaining it, I didn't quite understand how that worked, this will be my last "bump", unless valid to the conversation.

**Comment by ghost** [comment]
If the developers really just don't want to add this controversial but I think important feature, then fine, and sorry for trying to improve Godot. I originally thought that the devs were very open to improvement, but I feel like my ideas are not welcome, but no hard feelings anyway. This thread can now be closed or locked.

**Reply by Zireael07** [reply]
Man, you were told not to "bump" and then did it anyway after a day.

Godot has a policy of keeping the core very "lean". That said, I'm seeing interest here, and the fact that no one is rushing to do a PR for this after a couple of days does not mean your ideas aren't welcome

**Reply by ghost** [reply]
But you just "bumped", and arguably, I contributed significant new information, you did not.
Also, I am only seeing interest from users, but the devs don't seem to be listening to even them, so just forget it, weave me awone.
